lib/Interpreter/IncrementalCUDADeviceCompiler.h

```diff
@@ -157,49 +116,78 @@ namespace cling {
     ///\param [in] invocationOptions - The invocationOptions of the interpreter.
     ///\param [in] intprOptLevel - The optimization level of the interpreter.
     ///\param [in] debugInfo - The debugInfo of the CompilerInstance.
+    ///\param [in] hostTriple - The llvm triple of the host system
     void setCuArgs(const clang::LangOptions& langOpts,
                    const cling::InvocationOptions& invocationOptions,
                    const int intprOptLevel,
-                   const clang::codegenoptions::DebugInfoKind debugInfo);
-
-    ///\brief Save .cu file, if cuda device code compiler failed at translation.
-    ///
-    ///\returns The error code of the rename operation
-    std::error_code saveFaultyCUfile();
+                   const clang::codegenoptions::DebugInfoKind debugInfo,
```

Comment:
Why are these new parameters (and the `const int intprOptLevel` before) marked `const`?

Reply:
The `llvm::Triple` is necessary for multi-platform support. I could reduce the interface to `setCuArgs(const clang::CompilerInstance& CI, const cling::InvocationOptions& invocationOptions)` if it is better.

`const int intprOptLevel` is a relic of the old implementation. The old implementation did not allow to change the optimization level at runtime. The new implementation should allow this.  I will check it in a future commit.